src/main/java/seedu/address/model/Model.java

```diff
@@ -113,4 +113,30 @@ public interface Model {
      * @throws NullPointerException if {@code person} is null.
      */
     void updateSinglePersonList(Person person);
+
```

Comment:
Good application of separation of concerns principles! The navigation between states of the addressbook (be it  undo or redo) are split into distinct sections.